src/native/minipal/debugger.c

```diff
@@ -17,28 +17,43 @@
 
 #if defined(_WIN32)
 #include <windows.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(__linux__)
 #include <stdio.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(__APPLE__) || defined(__FreeBSD__)
 #include <sys/sysctl.h>
 #include <sys/types.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(__NetBSD__)
 #include <kvm.h>
 #include <sys/param.h>
 #include <sys/sysctl.h>
 #include <sys/proc.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(__sun)
 #include <fcntl.h>
 #include <procfs.h>
 #include <errno.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(_AIX)
 #include <sys/proc.h>
 #include <sys/types.h>
 #include <sys/procfs.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #elif defined(__HAIKU__)
 #include <OS.h>
+#define MINIPAL_DEBUGGER_PRESENT_CHECK
 #endif
 
+bool minipal_can_check_for_native_debugger(void)
+{
+#if defined(MINIPAL_DEBUGGER_PRESENT_CHECK)
```

Comment:
Looks like an always-true condition?

Reply:
It’s true for every platform we support, but if you build on an unsupported platform, we’ll hit the false case.

Reply:
IMO, we should just delete `minipal_can_check_for_native_debugger` and add `#else\n#error "Not yet implemented"` on line 47 to let the person/team consciously make a determination during the porting time (whether they way to ignore this feature by adding `#elif defined(__someOS__) /* ignored */` or implement it as part of the port).

Reply:
IMO, just delete `minipal_can_check_for_native_debugger` and add `#else\n#error "Not yet implemented"` on line 47 to let the person/team consciously make a determination during the porting time (whether they want to ignore this feature by adding `#elif defined(__someOS__) /* ignored */` or implement it as part of the port).

Reply:
@AaronRobinsonMSFT, without the additional API: https://github.com/dotnet/runtime/compare/main...am11:runtime:dedup-debugger-check, it wouldn't block HelloWorld during the port to new platform. To get by, the person will add an empty elif branch likely with a TODO (as they would need to adjust tons of other places) and obviously not use `-d` or `--debug` options with `corerun`, as `wait_for_debugger` is only called in that case. 

No big deal, but that build-time error is the more prevalent pattern used in coreclr (I remembered hitting them in past adventures 😅).